packages/amplify_authenticator/example/integration_test/utils/data_utils.dart

```diff
@@ -98,3 +102,62 @@ Future<AdminCreateUserResponse?> adminCreateUser(
     }
   });
 }
+
+Future<ConfirmationCodeTestRun> subscribeToOTPCode({
+  Future<void> Function()? onSubscriptionEstablished,
+  Future<void> Function(String code)? onCodeRecieved,
+}) async {
+  Completer<ConfirmationCodeTestRun> codeCompleter =
+      Completer<ConfirmationCodeTestRun>();
+  String subscriptionDocument = '''subscription MyMutation {
+          onCreateConfirmSignUpTestRun {
+            id
+          }
+        }''';
+
+  String queryDocument = '''query GetCode(\$id: ID!) {
+        getConfirmSignUpTestRun(id: \$id) {
+          id
+          currentCode
+        }
+      }''';
+
+  Completer<void> subscriptionCompleter = Completer<void>();
+
+  final Stream<GraphQLResponse<String>> operation = Amplify.API.subscribe(
+    GraphQLRequest<String>(document: subscriptionDocument),
+    onEstablished: () {
+      subscriptionCompleter.complete();
```

Comment:
I assume you tested and this worked?  When I tried this approach it hung.

Reply:
It hung some (most?) of the time but did seem to work some of the time. I have `await subscriptionCompleter.future` commented out and instead have `await Future.delayed(const Duration(seconds: 1), () {});` as a temp solution.

Reply:
Alrighty, thanks